### DISCUSSION TITLE
请问Xray(E+F+H+A)+Caddy(N)中NaiveProxy使用QUIC协议传输连接不通，是配置问题还是我客户端问题？

### DISCUSSION THREAD

**Discussion by zjy7723** [lxhao61/integrated-examples#210] [Q&A] [answered]
我的使用环境：win10, klzgrad/naiveproxy 官方客户端
按照Xray(E+F+H+A)+Caddy(N)配置文件2_caddy.json 设置
服务器运行正常
使用官方客户端 用h3.xx.yy域名 QUIC协议连接不通。
我在配置文件2_caddy.json 中194行 cipher_suites 下添加"alpn": ["h2","http/1.1"]
再用h3.xx.yy域名QUIC协议测试  这时连接才通。
请问是配置问题还是我客户端问题？

**Comment by lxhao61** [comment, marked as answer]
我参考你的思路添加了 ``"alpn": ["h3","h2","http/1.1"]`` 测试，确实让使用 QUIC 协议连接成功率得到了改善，与使用 HTTPS 协议一样了。
一直以来我测试使用  QUIC 协议连接也不稳定，经常无法连接成功；还以为是网络问题及 QUIC 不成熟固有问题，现在来看应该是 NaiveProxy 使用 QUIC 协议交互 BUG 了。你可去 NaiveProxy 提 ``PR``，告知这个情况。
另外在没有解决这个问题之前要求在服务端配置 ``"alpn": ["h3","h2","http/1.1"]``，指定交互优先级。默认无需指定，协商即可。

**Reply by zjy7723** [reply]
谢谢你的测试解答。

**Reply by lxhao61** [reply]
你是在电脑上使用 NaiveProxy 客户端吗？需要你配合测试一下。

**Reply by zjy7723** [reply]
我目前是Debian系统TProxy透明代理网关，在Debian系统用docker 跑的NaiveProxy。我电脑上也可以跑 NaiveProxy 客户端，需要我测试什么请讲。

**Reply by lxhao61** [reply]
你服务端去除 ``"alpn": ["h3","h2","http/1.1"]`` 配置后重启，然后在电脑上跑 NaiveProxy 客户端测试你报告情况，NaiveProxy 客户端配置参考如下：
```
{
  "listen": "socks://127.0.0.1:1080",
  "proxy": "https://user:pass@example.com",
  "log": "",
  "log-net-log": "D:\\naiveproxy\\netlog.log"
}
```
[取得 netlog 看一下问题](https://github.com/klzgrad/naiveproxy/issues/657)，你提交一下。
备注：``D:\\naiveproxy\\netlog.log`` 为 netlog 日志路径，根据实际情况修改。

**Reply by zjy7723** [reply]
目前我服务端关闭了udp,前几天测试开启udp，发现服务端cpu有时会很高。过几天我再测试一下把netlog.log贴上来。

**Reply by lxhao61** [reply]
好。
